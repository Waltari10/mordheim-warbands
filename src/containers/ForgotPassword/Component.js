import React, { useState } from 'react';
import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';
import Grid from '@material-ui/core/Grid';
import Link from '@material-ui/core/Link';
import InputAdornment from '@material-ui/core/InputAdornment';
import { makeStyles } from '@material-ui/core/styles';
import { Link as RouterLink, Redirect } from 'react-router-dom';
import EmailIcon from '@material-ui/icons/EmailOutlined';
import InputBase from '@material-ui/core/InputBase';
import * as constants from '../../constants';

import AppWindow from '../../components/AppWindow';

const useStyles = makeStyles((theme) => ({
  header: {
    marginTop: theme.spacing(0),
    marginBottom: theme.spacing(12),
  },
  email: {
    background: 'white',
    borderTopRightRadius: '4px',
    borderTopLeftRadius: '4px',
    height: '48px',
    overflow: 'hidden',
    borderWidth: '0.5px',
    borderStyle: 'solid',
    borderColor: '#C0C0C0',
  },
  continueButton: {
    marginTop: theme.spacing(3),
    marginBottom: theme.spacing(3),
  },
  loginLink: {
    marginBottom: theme.spacing(3),
  },
  content: {
    width: '100%',
  },
  errorMsg: {
    marginBottom: theme.spacing(3),
  },
  inputRoot: {
    backgroundColor: 'white',
    height: '48px',
  },
  input: {
    height: '48px',
    padding: '27px 12px 24px',
  },
  inputAdornment: {
    color: theme.palette.primary.main,
  },
  inputStartAdornment: {
    color: theme.palette.primary.main,
    marginLeft: theme.spacing(2),
  },
}));


const Login = ({
  isAuthorized, sendResetPasswordEmailRequestState, sendResetPasswordEmail,
}) => {

  const [email, setEmail] = useState('');

  const isDisabled = (
    sendResetPasswordEmailRequestState === constants.LOADING ||
    sendResetPasswordEmailRequestState === constants.SUCCESS
  );

  const submitDisabled = isDisabled || !email;

  const classes = useStyles();

  return (
    <AppWindow size="xs">

      {
        isAuthorized &&
          <Redirect
            to='/'
          />
      }

      <Grid className={classes.content} alignItems="center" direction="column" container>
        <Grid className={classes.header} item>
          <Typography variant="h4">Mordheim Warbands</Typography>
        </Grid>
        <Grid item>
          <InputBase
            style={{ display: 'flex' }}
            className={classes.email}
            onChange={(e) => setEmail(e.target.value)}
            // label="Type your email"
            disabled={isDisabled}
            placeholder="email"
            value={email}
            variant="filled"
            startAdornment={
              <InputAdornment position="start">
                <EmailIcon className={classes.inputStartAdornment} />
              </InputAdornment>
            }
          />
        </Grid>
        <Grid item>
          <Button
            className={classes.continueButton}
            disabled={submitDisabled}
            onClick={() => sendResetPasswordEmail(email)}
            variant="contained"
            color="primary"
          >Send password reset email</Button>
        </Grid>

        {
          sendResetPasswordEmailRequestState === constants.ERROR && (
            <Grid className={classes.errorMsg} item>
              <Typography color="error" variant="body1">Something went wrong!</Typography>
            </Grid>
          )
        }
        <Grid
          className={classes.loginLink}
          item
        >
          <Link
            component={RouterLink}
            to={'/'}
          >
            <Typography variant="body2">
              Go back to login
            </Typography>
          </Link>
        </Grid>
      </Grid>
    </AppWindow>
  );
};
 
export default Login;