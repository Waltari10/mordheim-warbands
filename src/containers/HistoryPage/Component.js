import React from 'react';

// TODO: Add indicator indicating how far user is in reflcetion
// Add 5xwhy


const HistoryPage = () => {
  return (<div>HistoryPage</div>);
};
 
export default HistoryPage;