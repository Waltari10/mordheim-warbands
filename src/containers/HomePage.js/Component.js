import React, { useEffect } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Progress from '@material-ui/core/CircularProgress';
import Fab from '@material-ui/core/Fab';
import AddIcon from '@material-ui/icons/Add';
import Menu from '@material-ui/core/Menu';
import { Link as RouterLink } from 'react-router-dom';
import MenuItem from '@material-ui/core/MenuItem';
import MenuIcon from '@material-ui/icons/Menu';
import IconButton from '@material-ui/core/IconButton';
import { Typography } from '@material-ui/core';

import PrivateRoute from '../PrivateRoute';
import ReflectionModal from '../ReflectionModal';

import ReflectionListItem from './ReflectionListItem';
import AppWindow from '../../components/AppWindow';


const useStyles = makeStyles((theme) => ({
  listItem: {
    
  },
  window: {
    padding: 0,
    position: 'relative',
    flexDirection: 'column',
    overflow: 'hidden',
  },
  container: {
    marginTop: '53px',
    width: '100%',
    height: '100%',
    placeSelf: 'flex-start',
    overflowY: 'scroll',
    overflowX: 'hidden',
    flex: 1,
  },
  progress: {
    position: 'absolute',
    top: 'calc(50% - 10px)',
    left: 'calc(50% - 10px)',
  },
  hero: {
    position: 'absolute',
    right: theme.spacing(3),
    bottom: theme.spacing(3),
  },
  addIcon: {
    color: 'white',
    marginTop: '6px',
  },
  topNavigationContainer: {
    boxShadow: '0px 3px 3px -2px rgba(0,0,0,0.2), 0px 3px 4px 0px rgba(0,0,0,0.14), 0px 1px 8px 0px rgba(0,0,0,0.12)',
    height: '53px',
    position: 'absolute',
    width: '100%',
    zIndex: 1,
    left: 0,
    right: 0,
    backgroundColor: '#9e9e9e',
    top: 0,
  },
  title: {
    lineHeight: '53px',
  },
  menuIcon: {

    position: 'absolute',
    top: '5px',
    right: theme.spacing(3),
  },
}));


const HomePage = ({
  isLoading, reflections = {}, getReflections, match, logout, uid,
  
}) => {

  useEffect(() => {
    if (uid) {
      getReflections();
    }
  }, [uid]);


  const classes = useStyles();

  const [anchorEl, setAnchorEl] = React.useState(null);

  const handleClick = event => {
    setAnchorEl(event.currentTarget);
  };

  const handleClose = () => {
    setAnchorEl(null);
  };

  return (
    <AppWindow className={classes.window} size="xs">

      <div
        className={classes.topNavigationContainer}
      >
        <Typography className={classes.title} align="center" variant="h6">
            Reflection log
        </Typography>


        <IconButton
          className={classes.menuIcon}
          onClick={handleClick}
        >
          <MenuIcon />
        </IconButton>

        <Menu
          id="simple-menu"
          anchorEl={anchorEl}
          keepMounted
          open={Boolean(anchorEl)}
          onClose={handleClose}
        >
          <MenuItem onClick={() => {
            handleClose();
            logout();
          }}>Logout</MenuItem>
        </Menu>

        
      </div>


      <div className={classes.container}>
        <PrivateRoute component={ReflectionModal} path="/reflection/:id" />
        {isLoading && <Progress className={classes.progress} />}
        {!isLoading && Object.keys(reflections).map((key) => {
          const reflection = reflections[key];

          return (
            <ReflectionListItem
              match={match}
              key={key}
              id={key}
              reflection={reflection}
            />
          );
        })}

        {!isLoading && Object.keys(reflections).length === 0 && (
          <Typography>No reflections yet. Add first reflection.</Typography>
        )}

        
      </div>


      <Fab
        className={classes.hero}
        color="primary"
      >
        <RouterLink
          to={`${match.path}reflection/new`}
        >
          <AddIcon className={classes.addIcon}/>
        </RouterLink>
      </Fab>
    </AppWindow>
  );
};
 
export default HomePage;