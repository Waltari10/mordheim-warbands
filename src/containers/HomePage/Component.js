import React, { useEffect } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Progress from '@material-ui/core/CircularProgress';
import Fab from '@material-ui/core/Fab';
import AddIcon from '@material-ui/icons/Add';
import Menu from '@material-ui/core/Menu';
import { Redirect } from 'react-router-dom';
import MenuItem from '@material-ui/core/MenuItem';
import MenuIcon from '@material-ui/icons/Menu';
import IconButton from '@material-ui/core/IconButton';
import { Typography } from '@material-ui/core';


import * as constants from '../../constants';
import PrivateRoute from '../PrivateRoute';
import WarbandModal from '../WarbandModal';

import WarbandListItem from './WarbandListItem';


const useStyles = makeStyles((theme) => ({
  fabProgress: {
    marginTop: '7px',
  },
  window: {
    padding: 0,
    position: 'relative',
    flexDirection: 'column',
    height: '100%',
    width: '100%',
    backgroundColor: theme.palette.background,
  },
  container: {
    margin: '0 auto',
    display: 'block',
    marginTop: '48px',
    width: '100%',
    maxWidth: '1024px',
    height: 'calc(100% - 48px)',
    placeSelf: 'flex-start',
    overflowY: 'scroll',
    overflowX: 'hidden',
    flex: 1,
  },
  progress: {
    position: 'absolute',
    top: 'calc(50% - 10px)',
    left: 'calc(50% - 10px)',
  },
  hero: {
    display: 'block',
    margin: '0 auto',
    marginTop: theme.spacing(6),
    marginBottom: theme.spacing(6),
  },
  topNavigationContainer: {
    boxShadow: '0px 3px 3px -2px rgba(0,0,0,0.2), 0px 3px 4px 0px rgba(0,0,0,0.14), 0px 1px 8px 0px rgba(0,0,0,0.12)',
    height: '48px',
    position: 'absolute',
    width: '100%',
    zIndex: 1,
    left: 0,
    right: 0,
    backgroundColor: theme.palette.primary.main,
    top: 0,
  },
  title: {
    lineHeight: '48px',
    color: 'white',
  },
  menuIcon: {
    position: 'absolute',
    top: '0px',
    right: theme.spacing(2),
  },
  addIcon: {
    color: 'white',
    marginTop: '7px',
  },
  menu: {
    backgroundColor: 'white',
  },
}));


const HomePage = ({
  isLoading, warbands = {}, getWarbands, match, logout, uid,
  warbandsIndex = [], addWarband, lastAddedWarbandId,
  addWarbandRequestState,
}) => {

  useEffect(() => {
    if (uid) {
      getWarbands();
    }
  }, [uid]);


  const classes = useStyles();

  const [anchorEl, setAnchorEl] = React.useState(null);

  const handleClick = event => {
    setAnchorEl(event.currentTarget);
  };

  const handleClose = () => {
    setAnchorEl(null);
  };

  return (
    <div className={classes.window}>

      {
        (addWarbandRequestState === constants.SUCCESS && lastAddedWarbandId) && (
          <Redirect
            to={`${match.path}warband/${lastAddedWarbandId}`}
          />)
      }

      <div
        className={classes.topNavigationContainer}
      >
        <Typography className={classes.title} align="center" variant="h5">
            Mordheim Warbands
        </Typography>


        <IconButton
          className={classes.menuIcon}
          onClick={handleClick}
        >
          <MenuIcon
            className={classes.whiteMenuIcon}
          />
        </IconButton>

        <Menu
          id="simple-menu"
          anchorEl={anchorEl}
          keepMounted
          open={Boolean(anchorEl)}
          onClose={handleClose}
          classes={{
            paper: classes.menu,
          }}
        >
          <MenuItem
            onClick={() => {
              handleClose();
              logout();
            }}>Logout</MenuItem>
        </Menu>


      </div>


      <div className={classes.container}>
        <PrivateRoute component={WarbandModal} path="/warband/:id" />
        {<Progress className={classes.progress} />}
        {warbandsIndex.map(key => {
          const warband = warbands[key];

          return (
            <WarbandListItem
              match={match}
              key={key}
              id={key}
              warband={warband}
            />
          );
        })}


        {!isLoading && (
          <Fab
            className={classes.hero}
            disabled={addWarbandRequestState === constants.LOADING}
            color="primary"
            onClick={() => {
              addWarband({});
            }}
          >
            {
              addWarbandRequestState === constants.LOADING ?
                <Progress className={classes.fabProgress}/> :
                <AddIcon className={classes.addIcon} />
            }
          </Fab>
        )}

        {!isLoading && Object.keys(warbands).length === 0 && (
          <Typography>No Warbands yet. Add first warband.</Typography>
        )}
      </div>
    </div>
  );
};

export default HomePage;