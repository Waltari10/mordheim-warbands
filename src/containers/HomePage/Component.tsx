import React, { useEffect, useState } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Progress from '@material-ui/core/CircularProgress';
import Fab from '@material-ui/core/Fab';
import AddIcon from '@material-ui/icons/Add';
import Menu from '@material-ui/core/Menu';
import { match as Match, Redirect } from 'react-router-dom';
import MenuItem from '@material-ui/core/MenuItem';
import MenuIcon from '@material-ui/icons/Menu';
import IconButton from '@material-ui/core/IconButton';
import { Theme, Typography } from '@material-ui/core';

import CreateWarbandDialog from './components/CreateWarbandDialog';
import * as constants from '../../constants';
import PrivateRoute from '../PrivateRoute';
import WarbandModal from '../WarbandModal/index';
import SupportDialog from '../../components/SupportContact/index';

import WarbandListItem from './WarbandListItem';
import { DispatchProps, StateProps } from './Container';


const useStyles = makeStyles((theme: Theme) => ({
  fabProgress: {
    marginTop: '7px',
  },
  window: {
    padding: 0,
    position: 'relative',
    flexDirection: 'column',
    height: '100%',
    width: '100%',
    backgroundColor: theme.palette.background.default,
  },
  container: {
    margin: '0 auto',
    display: 'block',
    marginTop: '48px',
    width: '100%',
    maxWidth: '1024px',
    height: 'calc(100% - 48px)',
    placeSelf: 'flex-start',
    overflowY: 'auto',
    overflowX: 'hidden',
    flex: 1,
  },
  progress: {
    position: 'absolute',
    top: 'calc(50% - 10px)',
    left: 'calc(50% - 10px)',
  },
  hero: {
    display: 'block',
    margin: '0 auto',
    marginTop: theme.spacing(6),
    marginBottom: theme.spacing(6),
  },
  topNavigationContainer: {
    boxShadow: '0px 3px 3px -2px rgba(0,0,0,0.2), 0px 3px 4px 0px rgba(0,0,0,0.14), 0px 1px 8px 0px rgba(0,0,0,0.12)',
    height: '48px',
    position: 'absolute',
    width: '100%',
    zIndex: 1,
    left: 0,
    right: 0,
    backgroundColor: theme.palette.primary.main,
    top: 0,
  },
  title: {
    lineHeight: '48px',
    color: 'white',
    marginLeft: '48px',
    marginRight: '48px',
    width: 'calc(100% - 96px)',
    textOverflow: 'ellipsis',
    overflow: 'hidden',
    whiteSpace: 'nowrap',
  },
  menuIcon: {
    position: 'absolute',
    top: '0px',
    right: theme.spacing(2),
  },
  whiteMenuIcon: {
    color: 'white',
  },
  addIcon: {
    color: 'white',
    marginTop: '7px',
  },
  menu: {
    backgroundColor: 'white',
  },
  addFirstWarband: {
    marginTop: theme.spacing(12),
  },
}));

interface Props {
  match: Match;
}


const HomePage: React.FunctionComponent<StateProps & DispatchProps & Props> = ({
  isLoading, warbands = {}, getWarbands, match, logout, uid,
  warbandsIndex = [], addWarband, lastAddedWarbandId,
  addWarbandRequestState,
}) => {

  useEffect(() => {
    if (uid) {
      getWarbands();
    }
  }, [uid]);


  const classes = useStyles();

  const [anchorEl, setAnchorEl] = React.useState(null);

  const handleClick = event => {
    setAnchorEl(event.currentTarget);
  };

  const handleClose = () => {
    setAnchorEl(null);
  };


  const [isSupportOpen, setIsSupportOpen] = useState(false);
  const [isCreateWarbandOpen, setIsCreateWarbandOpen] = useState(false);

  return (
    <div className={classes.window}>

      <CreateWarbandDialog
        addWarband={addWarband}
        open={isCreateWarbandOpen}
        close={() => setIsCreateWarbandOpen(false)}
      />

      {
        (addWarbandRequestState === constants.SUCCESS && lastAddedWarbandId) && (
          <Redirect
            to={`${match.path}warband/${lastAddedWarbandId}`}
          />)
      }

      <div
        className={classes.topNavigationContainer}
      >
        <Typography id="homepage_header" className={classes.title} align="center" variant="h5">
          Warbands
        </Typography>


        <IconButton
          className={classes.menuIcon}
          onClick={handleClick}
          id="homepage_menu_button"
        >
          <MenuIcon
            className={classes.whiteMenuIcon}
          />
        </IconButton>

        <Menu
          id="simple-menu"
          anchorEl={anchorEl}
          keepMounted
          open={Boolean(anchorEl)}
          onClose={handleClose}
          classes={{
            paper: classes.menu,
          }}
        >
          <MenuItem
            id="homepage_logout_button"
            onClick={() => {
              handleClose();
              logout();
            }}>Logout</MenuItem>
          <MenuItem
            onClick={() => {
              setIsSupportOpen(true);
            }}>Support/contact</MenuItem>
          {/* <MenuItem
            onClick={() => {
              setIsDisclaimerOpen(true);
            }}>Disclaimer</MenuItem> */}
        </Menu>


      </div>

      <SupportDialog
        open={isSupportOpen}
        handleClose={() => {
          setIsSupportOpen(false);
        }}
      />

      <div className={classes.container}>
        <PrivateRoute component={WarbandModal} path="/warband/:id" />
        {isLoading && <Progress className={classes.progress} />}
        {warbandsIndex.map(key => {
          const warband = warbands[key];

          return (
            <WarbandListItem
              match={match}
              key={key}
              id={key}
              warband={warband}
            />
          );
        })}

        {!isLoading && Object.keys(warbands).length === 0 && (
          <Typography
            variant="h6"
            className={classes.addFirstWarband}
            align="center"
          >No Warbands yet. Add first warband.</Typography>
        )}


        {!isLoading && (
          <Fab
            id="add_warband_button"
            className={classes.hero}
            disabled={addWarbandRequestState === constants.LOADING}
            color="primary"
            onClick={() => {
              setIsCreateWarbandOpen(true);
            }}
          >
            {
              addWarbandRequestState === constants.LOADING ?
                <Progress className={classes.fabProgress} /> :
                <AddIcon className={classes.addIcon} />
            }
          </Fab>
        )}

      </div>
    </div>
  );
};

export default HomePage;