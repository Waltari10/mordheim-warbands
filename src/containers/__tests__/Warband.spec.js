
import React from 'react';
import { render } from '@testing-library/react';
import { HashRouter as Router } from 'react-router-dom';
// eslint-disable-next-line no-unused-vars
import { createMuiTheme, MuiThemeProvider } from '@material-ui/core/styles';

import Component from '../Warband/Component';


const theme = createMuiTheme({});


describe('Warband', () => {
  it('Should render (smoke test)', () => {
    const { asFragment } = render(
      // Requires to be wrapped in router because NotFoundPage uses react-router-doms Link element
      // Requires to be wrapped MuiThemeProvider because NotFoundPage is using Theme from material-ui
      <Router>
        <MuiThemeProvider theme={theme}>
          <Component
            addWarbandReset={() => {}}
          />
        </MuiThemeProvider>
      </Router>
    );
    const firstRender = asFragment();
    expect(firstRender).toMatchSnapshot();
  });
});