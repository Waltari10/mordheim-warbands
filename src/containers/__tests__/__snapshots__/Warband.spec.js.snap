// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Warband Should render (smoke test) 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-viewContainer-8"
  >
    <div
      class="makeStyles-form-7"
    >
      <div
        class="MuiPaper-root makeStyles-innerForm-6 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <h5
          class="MuiTypography-root makeStyles-title-12 MuiTypography-h5 MuiTypography-alignCenter"
        >
          No name
        </h5>
        <button
          class="MuiButtonBase-root MuiIconButton-root makeStyles-menuIcon-20"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              style="color: white;"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <div>
          <h5
            class="makeStyles-h5-1"
            id="general_header"
            variant="h5"
          >
            General
          </h5>
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
          >
            <div
              class="MuiGrid-root MuiGrid-item"
            >
              <div
                class="MuiFormControl-root MuiTextField-root makeStyles-textFieldLong-16"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                  data-shrink="false"
                >
                  Warband name
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input"
                    name="name"
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="PrivateNotchedOutline-root-294 MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="PrivateNotchedOutline-legendLabelled-296"
                    >
                      <span>
                        Warband name
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item"
            >
              <div
                class="MuiFormControl-root MuiTextField-root makeStyles-textFieldLong-16"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                  data-shrink="false"
                >
                  Warband type
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input"
                    name="type"
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="PrivateNotchedOutline-root-294 MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="PrivateNotchedOutline-legendLabelled-296"
                    >
                      <span>
                        Warband type
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item"
            >
              <div
                class="MuiFormControl-root MuiTextField-root makeStyles-numberField-14 makeStyles-textFieldShort-15"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
                  data-shrink="true"
                >
                  Games played
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input"
                    name="gamesPlayed"
                    type="number"
                    value="0"
                  />
                  <fieldset
                    aria-hidden="true"
                    class="PrivateNotchedOutline-root-294 MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="PrivateNotchedOutline-legendLabelled-296 PrivateNotchedOutline-legendNotched-297"
                    >
                      <span>
                        Games played
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </div>
          <hr
            class="MuiDivider-root"
            style="margin-top: 24px;"
          />
          <h5
            class="makeStyles-h5-1"
            id="wealth_header"
            style="padding-top: 24px;"
            variant="h5"
          >
            Wealth
          </h5>
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
          >
            <div
              class="MuiGrid-root MuiGrid-item"
            >
              <div
                class="makeStyles-textFieldLong-16"
              >
                <div
                  class="MuiFormControl-root MuiTextField-root makeStyles-numberField-14"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
                    data-shrink="true"
                  >
                    Wyrdstone shards
                  </label>
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiOutlinedInput-input"
                      name="shards"
                      type="number"
                      value="0"
                    />
                    <fieldset
                      aria-hidden="true"
                      class="PrivateNotchedOutline-root-294 MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="PrivateNotchedOutline-legendLabelled-296 PrivateNotchedOutline-legendNotched-297"
                      >
                        <span>
                          Wyrdstone shards
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
                <div
                  class="MuiFormControl-root MuiTextField-root makeStyles-goldCrowns-24 makeStyles-numberField-14"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
                    data-shrink="true"
                  >
                    Gold crowns
                  </label>
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiOutlinedInput-input"
                      name="goldCrowns"
                      type="number"
                      value="0"
                    />
                    <fieldset
                      aria-hidden="true"
                      class="PrivateNotchedOutline-root-294 MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="PrivateNotchedOutline-legendLabelled-296 PrivateNotchedOutline-legendNotched-297"
                      >
                        <span>
                          Gold crowns
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item"
            >
              <div
                class="MuiFormControl-root MuiTextField-root makeStyles-textFieldArea-17"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                  data-shrink="false"
                >
                  Equipment
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-multiline MuiOutlinedInput-multiline"
                >
                  <textarea
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
                    name="equipment"
                    rows="1"
                    style="height: 0px; overflow: hidden;"
                  />
                  <textarea
                    aria-hidden="true"
                    class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
                    readonly=""
                    style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); width: 100%;"
                    tabindex="-1"
                  />
                  <fieldset
                    aria-hidden="true"
                    class="PrivateNotchedOutline-root-294 MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="PrivateNotchedOutline-legendLabelled-296"
                    >
                      <span>
                        Equipment
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </div>
          <hr
            class="MuiDivider-root"
            style="margin-top: 24px;"
          />
          <h5
            class="makeStyles-h5-1"
            id="rating_header"
            style="padding-top: 24px;"
          >
            Rating
          </h5>
          <p
            class="MuiTypography-root makeStyles-textFieldShort-15 MuiTypography-body1"
          >
            Total experience: 0
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            Members (0) x 5: 0
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            Rating: 0
          </p>
          <hr
            class="MuiDivider-root"
            style="margin-top: 24px;"
          />
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-addHireButton-10 MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              <span
                class="MuiButton-startIcon MuiButton-iconSizeLarge"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                  />
                </svg>
              </span>
              Add hero
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <hr
            class="MuiDivider-root"
            style="margin-top: 24px;"
          />
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-addHireButton-10 MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              <span
                class="MuiButton-startIcon MuiButton-iconSizeLarge"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                  />
                </svg>
              </span>
              Add henchman
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      class="makeStyles-navigation-4"
    >
      <p
        class="MuiTypography-root MuiTypography-body1"
        style="margin-top: 16px; margin-bottom: 8px;"
      >
        <b>
          Navigation:
        </b>
      </p>
      <div
        class="makeStyles-navigationLink-5"
      >
        <a
          class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
        >
          → General
        </a>
      </div>
      <div
        class="makeStyles-navigationLink-5"
      >
        <a
          class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
        >
          → Wealth
        </a>
      </div>
      <div
        class="makeStyles-navigationLink-5"
      >
        <a
          class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
        >
          → Rating
        </a>
      </div>
      <p
        class="MuiTypography-root MuiTypography-body1"
        style="margin-top: 16px; margin-bottom: 8px;"
      >
        <b>
          Heroes:
        </b>
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1"
        style="margin-top: 16px; margin-bottom: 8px;"
      >
        <b>
          Henchmen:
        </b>
      </p>
    </div>
  </div>
</DocumentFragment>
`;
